tests/FSharp.Core.UnitTests/FSharp.Core/Microsoft.FSharp.Collections/MapModule.fs

```diff
@@ -61,6 +61,31 @@ type MapModule() =
         Assert.AreEqual(resultExt.[2], "dup")   
         
          
+        ()
+
+    [<Test>]
+    member this.Change() =
+
+        let a = (Map.ofArray [|(1,1);(2,4);(3,9)|])
+        let b = Map.change 4 (fun current -> Assert.AreEqual(current, None); Some 16) a
+        Assert.AreEqual(b.[4], 16)
```

Comment:
I'd add a check that the map still contains all the old pair as well (1, 2; 2, 4; 3, 9), because `change` function can, in theory, remove some or all of them.